src/pages/UpdateServers.js

```diff
@@ -155,14 +155,23 @@ class UpdateServers extends BaseUpdateWizardPage {
     let internalModelServers = this.state.internalModel.getIn(['internal', 'servers']).toJS();
     for (const server_id of serverIds.values()) {
       let server = internalModelServers.find(s => s.id == server_id);
+      if (!server) continue;
       try {
         const responseData = await fetchJson('/api/v2/monasca/server_status/' + server.hostname);
-        this.setState(prevState => {
-          let serverMonascaStatuses = prevState.serverMonascaStatuses;
-          serverMonascaStatuses[server_id] = translate('server.details.status.' + responseData.status);
-          return { 'serverMonascaStatuses' : serverMonascaStatuses };
+        this.setState({
+          serverMonascaStatuses: {
+            ...this.state.serverMonascaStatuses,
```

Comment:
I don't understand the point of this line. Why is it trying to set the state to the same value? And in general, referring to `this.state` with a `setState` call is a race condition, since the `setState` is executed asynchronously.

Reply:
This is setting the state variable `serverMonascaStatuses` to a new object with all the same properties as before (by copying them from `this.state.serverMonascaStatuses`) but with a new value for the one property with id value of `server_id` variable. This does not cause a "race condition", not possible in the browser we have only one thread of execution. Though something that we should be worried about is updating the state during a render, which can causes infinite re-renders.

Reply:
Upon reflect, my point #1 does not apply here because you are changing a value within an object within state, which `setState` will not detect, so that part is not a problem. You just need to pay attention to #2 by making use of `prevState`